### PR TITLE
update chat icons and add operator caching mechanism regarding ChatMe…

### PR DESCRIPTION
**Pull request by DavDo**
…ssage changes

MOB - 1542

**Jira issue:**
https://glia.atlassian.net/browse/MOB-1542

**Additional info:**

**Screenshots:**
image

### CODE REVIEW

File: widgetssdk/build.gradle [hunk 1]
```diff
@@ -55,7 +54,6 @@ dependencies {
     // https://square.github.io/leakcanary/changelog/
     // debugImplementation "com.squareup.leakcanary:leakcanary-android:$leakCanaryVersion"
 
-    // Replaced with a local AAR file
```
Comment (andrews-moc) [inline widgetssdk/build.gradle:58]: I think this line shouldn't be deleted

Comment (DavDo) [reply widgetssdk/build.gradle:58]: to be honest, I don't understand why we need this comment, in fact, dependency is not replaced with local aar.

File: widgetssdk/src/main/java/com/glia/widgets/chat/model/history/OperatorAttachmentItem.java [hunk 2]
```diff
@@ -7,51 +7,51 @@
...
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
+        if (!(o instanceof OperatorAttachmentItem)) return false;
         if (!super.equals(o)) return false;
         OperatorAttachmentItem that = (OperatorAttachmentItem) o;
-        return showChatHead == that.showChatHead &&
-                Objects.equals(attachmentFile, that.attachmentFile) &&
-                Objects.equals(operatorProfileImgUrl, that.operatorProfileImgUrl) &&
```
Comment (andrews-moc) [inline widgetssdk/src/main/java/com/glia/widgets/chat/model/history/OperatorAttachmentItem.java:35]: Just for my understanding, what is the reason for removing `Objects.equals(operatorProfileImgUrl, that.operatorProfileImgUrl)` ?

File: widgetssdk/src/main/java/com/glia/widgets/chat/model/history/OperatorAttachmentItem.java [hunk 3]
```diff
@@ -7,51 +7,51 @@
...
-                Objects.equals(isFileExists, that.isFileExists) &&
-                Objects.equals(isDownloading, that.isDownloading);
+        return isFileExists == that.isFileExists && isDownloading == that.isDownloading && Objects.equals(attachmentFile, that.attachmentFile);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(super.hashCode(), showChatHead, attachmentFile, operatorProfileImgUrl, isFileExists, isDownloading);
```
Comment (andrews-moc) [inline widgetssdk/src/main/java/com/glia/widgets/chat/model/history/OperatorAttachmentItem.java:42]: Just for my understanding, what is the reason for removing `showChatHead`, `operatorProfileImgUrl`?

File: widgetssdk/build.gradle (hunk 1 above)
Comment (DavDo) [reply widgetssdk/build.gradle:58]: we can add the local arr file and disable dependency manually, but this is only for test purposes on the local environment, that's why I found that line pointless.

File: widgetssdk/src/main/java/com/glia/widgets/chat/model/history/OperatorAttachmentItem.java (hunk 2 above)
Comment (DavDo) [reply widgetssdk/src/main/java/com/glia/widgets/chat/model/history/OperatorAttachmentItem.java:35]: The `operatorProfileImgUrl` field itself is removed and moved to the superclass

File: widgetssdk/build.gradle (hunk 1 above)
Comment (andrews-moc) [reply widgetssdk/build.gradle:58]: I see.

Maybe it makes sense to edit the comment?

I mean change "Replaced with a local AAR file" to something like "Disable this line if dependency shouldbe taken from local AAR file".

Although it's not related to the current feature. 🙂

**Review by andrews-moc — APPROVED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/chat/model/history/OperatorAttachmentItem.java (hunk 3 above)
Comment (DavDo) [reply widgetssdk/src/main/java/com/glia/widgets/chat/model/history/OperatorAttachmentItem.java:42]: moved to the `OperatorChatItem.java` class, which is the supertype for OperatorAttachmentItem and OperatorMessageItem.

**Review by AndriiHorishniiMOC — APPROVED** [review]

**Pull request merged**